feat: add suppressTrailingSpaces to remove all trailing spaces in final column

shreddedbacon opened the pull request:
## Proposed Changes
  - Add new `SuppressTrailingSpaces` that removes all the trailing spaces in the final column of a row. This is useful when using the `OptionsNoBordersAndSeparators` options to reduce the number of spaces at the end of the column.

shreddedbacon committed:
feat: add suppressTrailingSpaces to remove all trailing spaces in the final column

shreddedbacon commented:
I hope this is OK for me to submit. It isn't super important that it gets merged, I can deal with the trailing spaces. But it would be nice to be able to disable them as required for cleaner output.

jedib0t commented:
Can you show what happens when `OptionsNoBordersAndSeparators` is false and `SuppressLastColumnSpaces` is true?

jedib0t commented on table/writer.go:
```diff
@@ -33,6 +33,7 @@ type Writer interface {
 	SortBy(sortBy []SortBy)
 	Style() *Style
 	SuppressEmptyColumns()
+	SuppressLastColumnSpaces()
```
`SupressTrailingSpaces` may be a better name for this since this method is at a Table level.

shreddedbacon commented:
> Can you show what happens when `OptionsNoBordersAndSeparators` is false and `SuppressLastColumnSpaces` is true?

I added coverage in the tests to show how this looks when borders are turned on (they look bad as expected)

jedib0t commented:
> > Can you show what happens when `OptionsNoBordersAndSeparators` is false and `SuppressLastColumnSpaces` is true?
> 
> I added coverage in the tests to show how this looks when borders are turned on (they look bad as expected)

Yeah this code cannot be merged in this state. How about just adding a block of code after https://github.com/jedib0t/go-pretty/blob/main/table/table.go#L681? If the suppressTrailingSpaces is set to true, split the outStr into lines, and then trim each line before returning that?

shreddedbacon commented:
> > > Can you show what happens when `OptionsNoBordersAndSeparators` is false and `SuppressLastColumnSpaces` is true?
> > 
> > 
> > I added coverage in the tests to show how this looks when borders are turned on (they look bad as expected)
> 
> Yeah this code cannot be merged in this state. How about just adding a block of code after [`main`/table/table.go#L681](https://github.com/jedib0t/go-pretty/blob/main/table/table.go?rgh-link-date=2024-01-04T01%3A09%3A03Z#L681)? If the suppressTrailingSpaces is set to true, split the outStr into lines, and then trim each line before returning that?

Ahh, that works even better, thanks for the tip. And it retains the table shape when borders are present which is better than what I originally proposed.

~The sonarcloud code analysis complains about duplicated code in the `render_test` file though, most of it seems unrelated to my PR specifically here, happy to address if required though?~
I reduced the test cases down to eliminate what I think it was complaining about
